eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -147,6 +138,7 @@ jobs:
     CmakeArgs: ""
     AZURE_TEST_MODE: "LIVE"
     AZURE_LOG_LEVEL: "verbose"
+    VCPKG_CACHE_MODE: "readwrite"
```

Comment:
Why are you setting this explicitly here? Should it be handled by the template parameter? Also I assume readwrite means that you will need to be internal and have access to secrets to write.

Reply:
the way I see it, each service has a `ci.yaml` as the starting point.
the `ci.yaml` invokes `stages/archetype-sdk-client.yml` for both, `internal` and `public`.

Then, `stages/archetype-sdk-client.yml` uses `jobs/archetype-sdk-client.yml`, for both `internal` and `public`. It uses the stage name `Build`.  So, `build` runs public and internal.  Here, when invoking the `build` stage, I am checking `variables['System.TeamProject']` to either set `VcpkgCacheMode: 'readwrite'` or leave it with default value (read).  This covers the vcpkg cache for the build stage only, for both Internal and public.

Then, for internal pipelines, `stages/archetype-sdk-client.yml` also add stage `LiveTest` (jobs/archetype-sdk-tests.yml). Since this is **only** internal, I set `readwrite` directly as a variable in the matrix definition so it becomes an env var.  We don't need to make a input parameter for the stage here because it is always for internal only.

Finally, the `build` stage has one extra job ( out of the generated matrix ) called `GenerateReleaseArtifacts` which defines its own pool and variables.  So I also need to set the cache mode to the `${{ parameters.VcpkgCacheMode }}` so it only reads on public and read/write on internal pipeline.

As an alternative, I could use `readwrite` for everything, and let the public pipelines try to push cache to VCPKG and fail as the SAS secret won't be set.  VCPKG will just write to console something like `failed to push binary to the servers`, but woudn't make the pipeline to fail.  But this approach seems a little lazy, and someone would think that it is actually an error over there.

Reply:
Should we just remove the notion of the mode everywhere and simply check if we have a SAS token then set mode to readwrite and if not just read.

Then we ifdef the access of the SAS token on if the TeamProject is internal.

Reply:
Yes, I like the idea! :)